go/state/gostate/archive_state.go

```diff
@@ -155,14 +155,19 @@ func (s *ArchiveState) Export(ctx context.Context, out io.Writer) (common.Hash,
 		return common.Hash{}, state.ExportNotSupported
 	}
 
-	exportableTrie := mptio.NewExportableArchiveTrie(trie, s.block)
-	rootHash, err := mptio.ExportLive(ctx, mptio.NewLog(), exportableTrie, out)
+	err := mptio.ExportBlockFromOnlineArchive(ctx, nil, trie, out, s.block)
 	if err != nil {
 		s.archiveError = errors.Join(s.archiveError, err)
 		return common.Hash{}, s.archiveError
 	}
 
-	return rootHash, s.archiveError
+	rootHash, err := trie.GetHash(s.block)
+	if err != nil {
+		s.archiveError = errors.Join(s.archiveError, err)
+		return common.Hash{}, s.archiveError
+	}
+
+	return rootHash, nil
```

Comment:
I think in will be always nil (I changed this method a bit)

Reply:
It would just be defensive coding; It would not hurt to keep returning `s.archiveError`, even if it is always `nil`. Somebody may change something in this function in the future and overlook that in the end `nil` is returned.

Reply:
ok, makes sense, will change it this way